src/Agent.Listener/MessageListener.cs

```diff
@@ -129,12 +146,15 @@ public sealed class MessageListener : AgentService, IMessageListener
             ArgUtil.NotNull(Session, nameof(Session));
             ArgUtil.NotNull(_settings, nameof(_settings));
             var agentServer = HostContext.GetService<IAgentServer>();
+            int consecutiveErrors = 0; //number of consecutive exceptions thrown by GetAgentMessageAsync
```

Comment:
I have "no error", "one error" and "more than one error", which I cannot encode in a bool